### ISSUE TITLE
About object masks and training loss

### ISSUE THREAD

**Issue by Hank5120** [MarcelRogge/object-centric-2dgs#1]
Thank you for your great paper. 
I have a few questions regarding the object masks and training loss. 

1. Are the masks generated before the training the 2DGS or also used for the SFM estimates? 
2. Also, for the training loss, how the loss functions are integrated in the existing 2DGS pipeline?

**Comment by MarcelRogge** [comment]
1. In the paper, we generate masks only for the Mip-NeRF360 dataset and use the **already available SfM results, which did not have masks**. Generally, it is not necessary but possible to use the masks during SfM. I would assume the SfM to be more robust using the full images because more features can be found. However, if you would want to consider a special scenario such as a static camera and moving object, using the masks during SfM might be necessary to get a correct result.
2. First, we mask the inputs to the original photometric loss function (Eq.2 in our paper). 2DGS had already added 2 more losses to the photometric loss for a "total_loss". We also add the result of our background loss to this total loss (Eq.3 in our paper). There are no other integrations necessary for the loss functions.

Please let me know if this fully answers your questions.

**Comment by BigdateAB** [comment]
> Thank you for your great paper. I have a few questions regarding the object masks and training loss.
> 
> 1. Are the masks generated before the training the 2DGS or also used for the SFM estimates?
> 2. Also, for the training loss, how the loss functions are integrated in the existing 2DGS pipeline?

Did you modify the code in the paper? Could I request a copy of the modified code? Thank you.

**Comment by FeiyouG** [comment]
I implemented the object-centric 2DGS following the author's instructions. As the author mentioned, the process was straightforward. You can find my fork here: https://github.com/FeiyouG/2DGSx
